### DISCUSSION TITLE
Why do we need to pass `--bundle-output` even though the output is defined from within the webpack config?

### DISCUSSION THREAD

**Discussion by tom-sherman** [callstack/repack#166] [Q&A] [answered]
We specify `output.path` inside the webpack config, so why do we need to supply `--bundle-output` in to the `webpack-bundle` command?

Is it safe to supply a temporary directory here? If so, why not remove the need for the option in the CLI?

Related to #127

**Comment by zamotany** [comment, marked as answer]
`--bundle-output` is a destination for main bundle supplied by React Native when building with XCode or Gradle - without it we Re.Pack wouldn't know where to copy main bundle and the application `ipa`/`apk` would not include one, which would result in a crash. Additionally, if `--assets-dest` is not provided, the path will be inferred from `--bundle-output`.

When you're calling `react-native webpack-bundle` yourself, you could provide `--bundle-output <whatever>` and it will be all good. The reason why `--bundle-output` cannot be replaced with specifying `output.path`, is that on iOS the path is pointing into internal directories with UUID-like segments, so it's not easy to guess where it will be.

Another argument, is that `webpack-bundle` is modelled after `bundle` command from React Native CLI - we want `webpack-bundle` to have the same syntax and be a drop-in replacement.
